server/src/main/java/org/opensearch/common/util/concurrent/AbstractAsyncTask.java

```diff
@@ -89,6 +89,13 @@ public TimeValue getInterval() {
      */
     protected abstract boolean mustReschedule();
 
+    /**
+     * If the node is shutting down do not schedule this task.
+     */
+    protected boolean doNotRunWhenShuttingDown() {
```

Comment:
To be fair, I have difficulties to understand the complete picture here (my apologies for that), the mental model I have in mind is this:
 - all components should be closed on node shutdown
 - the async tasks that components maintain should also be closed on shutdown
 - the async task is not reschedule when closed


Taking `SegmentReplicationPressureService`, I would expect to observe:
  - Node::close()
  - SegmentReplicationPressureService::close()
  - AsyncFailStaleReplicaTask::close()

(same for PersistentTasksClusterService), what I am missing here?

Reply:
Thanks for asking for clarity - I've been making some assumptions and attempting to address them - thus those previous changes are not 'provable'.  I'm going to take a step back and add some `thread.sleeps` until I can manifest the issue locally.

I'll come back with some findings and address your workflow question.  It is a complex space.